File: android/app/src/main/java/com/mattermost/rnbeta/RNPasteableEditText.java
package com.mattermost.rnbeta;

import android.content.Context;

import com.facebook.react.views.textinput.ReactEditText;

public class RNPasteableEditText extends ReactEditText {

    private RNEditTextOnPasteListener mOnPasteListener;

    public RNPasteableEditText(Context context) {
        super(context);
    }

    public void setOnPasteListener(RNEditTextOnPasteListener listener) {
        mOnPasteListener = listener;
    }

    public RNEditTextOnPasteListener getOnPasteListener() {
        return mOnPasteListener;
    }
}

File: android/app/src/main/java/com/mattermost/rnbeta/RNPasteableEditText.kt
package com.mattermost.rnbeta

import android.content.Context
import com.facebook.react.views.textinput.ReactEditText

open class RNPasteableEditText(context: Context?) : ReactEditText(context) {
    private var mOnPasteListener: RNEditTextOnPasteListener? = null

    var onPasteListener: RNEditTextOnPasteListener?
        get() = mOnPasteListener
        set(listener) {
            mOnPasteListener = listener
        }
}
